### PR TITLE
runtimecfg: Add node-ip subcommand

### PR DESCRIPTION
**Pull request by celebdor**
node-ip is a subcommand that allows the user to see which IP should the
node use in cases of multiple interface and multiple address nodes. This
is useful to prevent cases where Container Runtime related services bind
to an interface that is not reachable in the control plane.

It has two subcommands:

* show: Takes one or more Virtual IPs of the control plane and it gives
  you one eligible IP on stdout.

* set: Takes one or more Virtual IPs of the control plane and sets
  systemd service configuration for services like CRI-O or Kubelet that
  need to bind to the control plane.

In the process of making the commit, and to make runtimecfg less of a
monolith, I split its subcommands to separate files.

### CODE REVIEW

**Comment by celebdor** [comment]
Moved the implementation from https://github.com/openshift/machine-config-operator/pull/1564

**Comment by celebdor** [comment]
/hold

Need to test a full deployment with the accompanying change in MCO that moves to use it.

**Comment by celebdor** [comment]
Need to add the retry on failure functionality from https://github.com/openshift/machine-config-operator/pull/1601 to prevent a regression

File: cmd/runtimecfg/node-ip.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	for {
+		nodeAddrs, err := utils.AddressesRouting(vips, utils.NonDeprecatedAddress, utils.NonDefaultRoute)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(nodeAddrs) > 0 {
+			chosen = nodeAddrs[0]
+			break
+		} else {
```
Comment (bcrochet) [inline cmd/runtimecfg/node-ip.go:140]: This may be a nit, but is the else necessary when the if breaks?

File: cmd/runtimecfg/node-ip.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+			if retry {
+				log.Errorf("Failed to find a suitable node IP")
+				time.Sleep(time.Second)
+			} else {
```
Comment (bcrochet) [inline cmd/runtimecfg/node-ip.go:144]: Can get rid of the else if you switch to if !retry { return ... }

**Comment by celebdor** [comment]
The binary works when copied to my IPv6 env:

`[core@master-0 ~]$ sudo ./runtimecfg node-ip set fd2e:6f44:5dd8:c956::2 
INFO[0000] Parsed Virtual IP fd2e:6f44:5dd8:c956::2     
INFO[0000] Checking whether address fe80::3438:32ff:fe83:e265/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::ace7:39ff:fe5d:174e/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fd00:1101::3/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::a2ec:1236:302:8010/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking route {Ifindex: 3 Dst: fd2e:6f44:5dd8:c956::/64 Src: <nil> Gw: <nil> Flags: [] Table: 254} (mask ffffffffffffffff0000000000000000) for address fd2e:6f44:5dd8:c956::14/128 
INFO[0000] Checking whether address fd2e:6f44:5dd8:c956::14/128 with route {Ifindex: 3 Dst: fd2e:6f44:5dd8:c956::/64 Src: <nil> Gw: <nil> Flags: [] Table: 254} contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Address fd2e:6f44:5dd8:c956::14/128 with route {Ifindex: 3 Dst: fd2e:6f44:5dd8:c956::/64 Src: <nil> Gw: <nil> Flags: [] Table: 254} contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::da45:8914:69a7:bc32/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::2c52:1fff:fe5e:455a/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::4c40:b9ff:fee9:a41e/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address 127.0.0.1/8 lo contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::286e:2dff:feb1:3220/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fd99::1/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::80dc:82ff:fe78:ba89/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::9c5b:dcff:fe76:18bd/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::ac65:aeff:fe29:c431/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::ac12:36ff:fee5:7260/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fd01:0:0:3::2/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::e849:dcff:fe33:d045/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::fcf1:85ff:feb8:4518/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::90ba:12ff:fe5d:fc3b/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::24ed:9cff:fe4d:3b2b/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Checking whether address fe80::c6f:3dff:fe6d:bfbe/64 contains VIP fd2e:6f44:5dd8:c956::2 
INFO[0000] Chosen Node IP fd2e:6f44:5dd8:c956::14       
INFO[0000] Opening Kubelet service override path /etc/systemd/system/kubelet.service.d/20-nodenet.conf 
INFO[0000] Writing Kubelet service override with content [Service]
Environment="KUBELET_NODE_IP=fd2e:6f44:5dd8:c956::14" 
INFO[0000] Opening CRI-O service override path /etc/systemd/system/crio.service.d/20-nodenet.conf 
INFO[0000] Writing CRI-O service override with content [Service]
Environment="CONTAINER_STREAM_ADDRESS=fd2e:6f44:5dd8:c956::14" 
[core@master-0 ~]$ ./runtimecfg node-ip show fd2e:6f44:5dd8:c956::2 2> /dev/null
fd2e:6f44:5dd8:c956::14
[core@master-0 ~]$ `

**Comment by celebdor** [comment]
verified to work on both masters and workers.

**Pull request merged**